lms/djangoapps/instructor/views/api.py

```diff
@@ -3397,6 +3400,28 @@ def update_cohort_assignment(request, course_id):
     return JsonResponse({})
 
 
+@transaction.non_atomic_requests
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_global_staff
+@require_http_methods(['POST', 'GET'])
+def cohorts_list_with_assignment(request, course_id):
+    cohort_info = json.loads(cohort_handler(request, course_id).content)
+    course_id = CourseKey.from_string(course_id)
+    cohort_assignments = (
+        CourseUserGroup.objects.filter(course_id=course_id).values_list('id', 'cohortassigment__user__email')
+    )
+
+    cohort_assignments_dict = defaultdict(set)
+
+    for key, value in cohort_assignments:
+        cohort_assignments_dict[key].add(value)
+
+    for cohort in cohort_info['cohorts']:
+        cohort['cohort_admins'] = list(cohort_assignments_dict.get(cohort['id'], []))
```

Comment:
Default list is not needed since we use defaultdict for the `cohort_assignments_dict`

Reply:
Make sense and was done by me.